### ISSUE TITLE
ncdfFlow installation 

### ISSUE THREAD

**Issue by haroon123** [RGLab/ncdfFlow#40]
I was trying to install ggcyto and realised that it needs ncdfFlow 2.17. I installed hdf5 library and tried to install ncdfFlow 2.17.5 ,

 R CMD INSTALL ncdfFlow_2.17.5.tar.gz --configure-args="--with-hdf5='/usr/local/lib'"

 it gave an error :

Error in dyn.load(file, DLLpath = DLLpath, ...) : 
  unable to load shared object '/home/haroon/R/x86_64-pc-linux-gnu-library/3.2/ncdfFlow/libs/ncdfFlow.so':
  /home/haroon/R/x86_64-pc-linux-gnu-library/3.2/ncdfFlow/libs/ncdfFlow.so: undefined symbol: H5P_CLS_DATASET_CREATE_ID_g
Error: loading failed
Execution halted
ERROR: loading failed
* removing ‘/home/haroon/R/x86_64-pc-linux-gnu-library/3.2/ncdfFlow’

I am not familiar with HDF file format and it is one of the reported bug in hdf5 library. Any help would be greatly appreciated.

I am on ubuntu 14.04, 64 bit machine, using R version 3.2 and hdf5-1.8.16

**Comment by mikejiang** [comment]
Make sure your `hdf5` library is in your searching path if it is not installed in the standard locations,
e.g. add your custom location to the path by adding this to `~/.profile` file
```bash
export LD_LIBRARY_PATH=/app/hdf5/1.8.13/lib:$LD_LIBRARY_PATH
```

**Comment by haroon123** [comment]
I have already added the path in ~/.bashrc so I think it should not make a difference. I updated the path in ~/.profile as well but still it gives the same error.

**Comment by SamGG** [comment]
Hi,
I am wondering why you did not install ncdfFlow package via the standard biocLite way. ncdfFlow package is available since a few releases, whereas ggcyto is not integrated with R 3.2. If it could help, here is the result of my install with Debian Mint. I guess  you did install the hdf5 devel package for compilation, but I remind this humbly.
HTH
```
* installing *source* package ‘ncdfFlow’ ...
configure: No directory was specified for --with-hdf5. Trying to find hdf5 using pkg-config.
checking for pkg-config... /usr/bin/pkg-config
Package hdf5 was not found in the pkg-config search path.
Perhaps you should add the directory containing `hdf5.pc'
to the PKG_CONFIG_PATH environment variable
No package 'hdf5' found
configure: pkg-config was not able to find the hdf5 library.
configure: Trying some common locations."
configure: search for hdf5 library in: /usr/local/lib
configure: search for hdf5 library in: /usr/local/lib64
configure: search for hdf5 library in: /usr/local/lib/x86_64-linux-gnu
configure: search for hdf5 library in: /usr/local/lib64/x86_64-linux-gnu
checking for h5cc... /usr/bin/h5cc
configure: PKG_CPPFLAGS =  
configure: PKG_LIBS = -lhdf5
configure: creating ./config.status
config.status: creating src/Makevars
** libs
g++ -I/usr/share/R/include -DNDEBUG   -I"/usr/local/lib/R/site-library/Rcpp/include" -I"/usr/local/lib/R/site-library/RcppArmadillo/include" -I"/usr/local/lib/R/site-library/BH/include"   -fpic  -O2 -pipe -g  -c RcppExports.cpp -o RcppExports.o
g++ -I/usr/share/R/include -DNDEBUG   -I"/usr/local/lib/R/site-library/Rcpp/include" -I"/usr/local/lib/R/site-library/RcppArmadillo/include" -I"/usr/local/lib/R/site-library/BH/include"   -fpic  -O2 -pipe -g  -c bitOps.cpp -o bitOps.o
gcc -std=gnu99 -I/usr/share/R/include -DNDEBUG   -I"/usr/local/lib/R/site-library/Rcpp/include" -I"/usr/local/lib/R/site-library/RcppArmadillo/include" -I"/usr/local/lib/R/site-library/BH/include"   -fpic  -O2 -pipe -g  -c hdfFlow.c -o hdfFlow.o
gcc -std=gnu99 -I/usr/share/R/include -DNDEBUG   -I"/usr/local/lib/R/site-library/Rcpp/include" -I"/usr/local/lib/R/site-library/RcppArmadillo/include" -I"/usr/local/lib/R/site-library/BH/include"   -fpic  -O2 -pipe -g  -c init.c -o init.o
g++ -I/usr/share/R/include -DNDEBUG   -I"/usr/local/lib/R/site-library/Rcpp/include" -I"/usr/local/lib/R/site-library/RcppArmadillo/include" -I"/usr/local/lib/R/site-library/BH/include"   -fpic  -O2 -pipe -g  -c readFrame.cpp -o readFrame.o
g++ -shared -L/usr/lib/R/lib -Wl,-z,relro -o ncdfFlow.so RcppExports.o bitOps.o hdfFlow.o init.o readFrame.o -lhdf5 -L/usr/lib/R/lib -lR
installing to /usr/local/lib/R/site-library/ncdfFlow/libs
** R
** inst
** preparing package for lazy loading
Creating a generic function for ‘lapply’ from package ‘base’ in package ‘ncdfFlow’
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded
* DONE (ncdfFlow)
```

**Comment by haroon123** [comment]
Exactly the same point. We can't install ncdfFlow 2.17.5 via biocLite which is the requirement for ggCyto as of now. Mike has written a very nice package, else I used to subset the data after gating and plot them using ggplot2 by customized functions. So I pushed it for manual install. Here is how I got it working :

sudo apt-get install libhdf5-dev
sudo apt-get install libhdf5-serial-dev
R CMD INSTALL ncdfFlow_2.17.5.tar.gz 
R CMD INSTALL flowCore_1.37.20.tar.gz
R CMD INSTALL flowWorkspace_3.17.43.tar.gz 
R CMD INSTALL ggcyto_0.99.8.tar.gz 

Seems I was doing something wrong while installing hdf5 manually, so I went for installation via apt-get.
